File: C#/proxyC#/Proxy.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace proxy
{
    class Proxy : ExampleInterface
    {
        private ExampleInterface exampleClass;
        private string password {get;set;}

        public Proxy(string password)
        {
            this.password = password;
        }

        private void CreateInstanceExampleClass()
        {
            exampleClass = new ExampleClass();
        }

        public void DoSomeThing()
        {
            if(password=="root")
            {
                CreateInstanceExampleClass();
                exampleClass.DoSomeThing();
            }
            else
            {
                Console.WriteLine("Wrong password");
            }
        }
    }
}

File: Java/ProxyJava/Proxy.java
import java.util.Scanner;


public class Proxy implements ExampleInterface {
	
	 private ExampleInterface exampleClass;
	 private String password;
	 
	 public Proxy(String password)
	 {
	     this.password = password;
	 }
	 
	 private void CreateInstanceExampleClass()
	 {
	     exampleClass = new ExampleClass();
     }
	 
	  public void DoSomeThing()
	  {
	     if("root".equals(password))
	     {
	         CreateInstanceExampleClass();
	         exampleClass.DoSomeThing();
	     }
	     else
	     {
	    	 System.out.print("Wrong password");
	     }
	 }
	  public static void main(String[] args){
		  Scanner password = new Scanner(System.in);
		  
		  Proxy proxy = new Proxy(password.nextLine());
		  proxy.DoSomeThing();
		  
		  password.close();
	  }

}


